### PR TITLE
Update AbstractNestablePropertyAccessor.java

### PR DESCRIPTION
**Pull request by ScOrPiOzzy**
https://github.com/spring-projects/spring-framework/issues/23937#issue-517809539

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@ScOrPiOzzy Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=23944)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-framework&pullRequestId=23944) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@ScOrPiOzzy Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=23944)!

File: spring-beans/src/main/java/org/springframework/beans/AbstractNestablePropertyAccessor.java [hunk 1]
```diff
@@ -779,7 +778,7 @@ private void growCollectionIfNecessary(Collection<Object> collection, int index,
 		}
 		int size = collection.size();
 		if (index >= size && index < this.autoGrowCollectionLimit) {
-			Class<?> elementType = ph.getResolvableType().getNested(nestingLevel).asCollection().resolveGeneric();
+			Class<?> elementType = ph.toTypeDescriptor().getResolvableType().getNested(nestingLevel).asCollection().resolveGeneric();
```
Comment (snicoll) [inline spring-beans/src/main/java/org/springframework/beans/AbstractNestablePropertyAccessor.java]: We haven't triaged the incoming issue yet and I can't say for sure if that change is needed but one way to make that more explicit would be to write some tests that fail without your change and pass with that change. We'd need tests for the issue you've raised.

**Review by snicoll — COMMENTED** [review]
Thanks for the PR. We haven't triaged the issue yet but I thought some feedback on your PR is useful anyway.

File: spring-beans/src/main/java/org/springframework/beans/AbstractNestablePropertyAccessor.java [hunk 2]
```diff
@@ -9,7 +9,6 @@
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
```
[the remark this answers is not in the document]
Comment (ScOrPiOzzy) [reply spring-beans/src/main/java/org/springframework/beans/AbstractNestablePropertyAccessor.java]: Sorry for my bad.

File: spring-beans/src/main/java/org/springframework/beans/AbstractNestablePropertyAccessor.java (hunk 1 above)
Comment (ScOrPiOzzy) [reply spring-beans/src/main/java/org/springframework/beans/AbstractNestablePropertyAccessor.java]: Hi, I submitted a test case
[https://github.com/spring-projects/spring-framework/pull/23944/commits/51ec366d870b97cfd648fdc18f2f25a0383064a6](https://github.com/spring-projects/spring-framework/pull/23944/commits/51ec366d870b97cfd648fdc18f2f25a0383064a6)

**Comment by izeye** [comment]
FYI, the referenced issue has been closed as invalid.

**Comment by snicoll** [comment]
Thanks. Closing as a follow-up of the issue.

**Pull request closed without merging**